src/mon/OSDMonitor.cc

```diff
@@ -984,6 +984,9 @@ void OSDMonitor::prime_pg_temp(
     return;  // can be no worse off than before
 
   if (next_up == next_acting) {
+    if (acting == next_acting) {
```

Comment:
I think we should group this with the comparison on line 977, or else the 977 check appears wrong/confusing on its own.
```
-  if (acting == next_acting && next_up != next_acting)
+  if (acting == next_acting &&
+      !(up != acting && next_up == next_acting))
     return;  // no change since last epoch
```
or similar

Reply:
(it's also the common case, so let's put it before we look up the pool etc.)